File: section3/ros2_ws/src/my_py_pkg/my_py_pkg/robot_news_station.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from example_interfaces.msg import String


class RobotNewsStationNode(Node): # MODIFY NAME
    def __init__(self):
        super().__init__("robot_news_station") # MODIFY NAME 

        # Explanation of the code below
        # self.publishers_ is a list of publishers.
        # Each publisher is responsible for publishing a message of type String to a topic.
        # The topic is called "robot_news".
        # The queue size is 10, which means that if messages are published faster than they are processed, the messages are buffered up to 10.
        self.robot_name_ = "C3PO"
        self.publishers_ = self.create_publisher(String, "robot_news", 10)

        # Set the timer
        # The timer is set to call the publish_news method every 0.5 seconds. (2Hz)
        self.timer_ = self.create_timer(0.5, self.publish_news)

        # Print the message for a best practice
        self.get_logger().info("Robot News Station has been started.")
    
    def publish_news(self):
        msg = String()
        msg.data = "Hi, this is " + self.robot_name_ + " from the Robot News Station."
        self.publishers_.publish(msg)
        


def main(args=None):
    rclpy.init(args=args)
    node = RobotNewsStationNode() # MODIFY NAME
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: section3/ros2_ws/src/my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class RobotNewsStationNode : public rclcpp::Node // MODIFY NAME
{
public:
    RobotNewsStationNode() : Node("robot_news_station"), robot_name_("R2D2")
    {
        publisher_ = this->create_publisher<example_interfaces::msg::String>("robot_news", 10);

        // Create Timer
        timer_ = this->create_wall_timer(std::chrono::milliseconds(500), std::bind(&RobotNewsStationNode::publishNews, this));

        // Print to console that the node is started
        RCLCPP_INFO(this->get_logger(), "Robot News Station has been started");
    }

private:
    void publishNews()
    {
        auto message = example_interfaces::msg::String();
        message.data = std::string("Hello, this is ") + robot_name_ + std::string(" from the robot news station");
        publisher_->publish(message);
    }

    std::string robot_name_;
    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;

    // Create Timer
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotNewsStationNode>(); // MODIFY NAME
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}
